main.cpp

```diff
@@ -210,285 +239,379 @@ struct Museum
 {
-    // 1) The high cut (or treble) setting (float)
     float trebleSetting = 0.5f;
-    // 2) The midrange cut setting (float)
     float midrangeSetting = 0.5f;
-    // 3) The low frequency (or bass) cut setting (float)
     float lowSetting = 0.5f;
-    // 4) the amount of out of phase post power amp feedback signal to mix in (presence) (float)
     float presenceSetting { 0.5f };
-    // 5) is the EQ tone stack bypassed? (bool)
     bool isBypassed = false;
 
-    //1) Adjust Treble control knob
     void adjustTrebleSetting(float value);
-    //2) Adjust Bass control knob
     void adjustBassSetting(float value);
-    //3) Turn bypass on/off
     void setBypass(bool value);
 };
-/*
-Thing 9) Speaker Cabinet
-5 properties:
-    1) the number of speakers (int)
-    2) the speaker diameter (int)
-    3) the cabinets max wattage (int)
-    4) the cabinet impedance in ohms (float)
-    5) the cabinets maximum reproduceable frequency (float)
-3 things it can do:
-    1) convert voltage and current to sound
-    2) adjust load attenuator
-    3) release magic smoke
- */
+
+void EQControls::adjustTrebleSetting(float value)
+{
+    trebleSetting = value;
+}
+
+void EQControls::adjustBassSetting(float value)
+{
+    lowSetting = value;
+}
+
+void EQControls::setBypass(bool value)
+{
+    isBypassed = value;
+}
+
 struct SpeakerCabinet
 {
-    // 1) the number of speakers (int)
     int numberOfSpeakers = 1;
-    // 2) the speaker diameter (int)
     int speakerDiameter = 12;
-    // 3) the cabinets max wattage (int)
     int maxWattage = 70;
-    // 4) the cabinet impedance in ohms (float)
     float impedance = 8.f;
-    // 5) the cabinets maximum reproduceable frequency (float)
     float maxFrequency = 6000.f;
 
-    // 1) convert voltage and current to sound
     double convertPowerToSound( float inputVoltage );
-    // 2) adjust load attenuator
     void setAttenuatorValue( float value );
-    // 3) release magic smoke
     void releaseMagicSmoke();
+private:
+    bool _speakersNotBlown = true;
 };
-/*
-Thing 10) Tube Guitar Amp Combo
-5 properties:
-    1) Power Supply
-    2) Output Section
-    3) Preamp Section
-    4) EQ Controls (or Tone Stack)
-    5) Speaker Cabinet
-3 things it can do:
-    1) Turn on / off
-    2) Adjust guitar EQ
-    3) Amplify guitar sound
- */
+
+double SpeakerCabinet::convertPowerToSound(float inputVoltage)
+{
+    if(_speakersNotBlown)
+    {
+        float wattageRequsted = inputVoltage * (inputVoltage / impedance);
+        double pascalsPerWatt = 2; // - using pascals because dB logarithmic stuff hard without framework to do math for me
+        if(wattageRequsted <= maxWattage * 1.1f)
+        {
+            // convert votage to assuming 2 pascals per watt 
+            return(static_cast<double>(inputVoltage) * (pascalsPerWatt * static_cast<double>(wattageRequsted)));
+        }
+        else
+        {
+            releaseMagicSmoke();
+            return 0;
+        }
+    }
+    return 0;
```

Comment:
D.R.Y. this `return 0` statement.
after you `releaseMagicSmoke`, you can exit the `if()` blocks.  the next line to execute will be the `return 0;` at the end of the function
- [ ] revise